content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+
+
+### Qt and Qt Resources
+
+The [Qt Resource System](https://doc.qt.io/qt-5/resources.html), which allows Gems to store and load image files via a `.qrc` file. This eliminates the need to load image files from absolute paths, making it simpler for you to distribute your Gem. Later, you will store an image file to create an icon for your tool.
+
+
+## Widgets and layouts 
+
+Widgets are containers for UI elements and layouts define how those UI elements are formatted. Your custom shape tool is essentially a main widget that contains sub-widgets of UI elements within it. Each widget can have its own layout. This nested widget and layout structure alllows you to organize groups of UI elements.
+
+The `ShapeExampleWidget.cpp` file already inherits from `QWidget`, which creates the main widget. Next, you must create a layout for the main widget. Refer to the snippet below for the following instructions: 
+
+1. At the top of your constructor, instantiate a `QVBoxLayout` called `mainLayout`.
+   
+2. Later, you will create various UI elements and add them to the `mainLayout`. 
+
+3. We recommend that you add a stretch at the bottom of `mainLayout` using `addStretch()` to fill any expanded space when you resize the window. 
+
+4. Finally, set `mainLayout` to be the layout for this widget using `setLayout()`. 
+
+```cpp
+ShapeExampleWidget::ShapeExampleWidget(QWidget* parent)
+    : QWidget(parent)
+{
+    setWindowTitle(QObject::tr("ShapeExample"));
+
+    QVBoxLayout* mainLayout = new QVBoxLayout(this);
+
+
+    // Add various UI elements to mainLayout
+
+
+    mainLayout->addStretch();
+
+    setLayout(mainLayout);
+}
+```
+
+## Input fields and check boxes
+
+In this step, you will create an input field for the entity's name and a check box to append a suffix that contains the component's name. For example, suppose you set the entity's name to "MyEntity" and enable the suffix check box. When you create Shape components of type Box and Sphere, they will respectively be named "MyEntity_BoxShape" and "MyEntity_SphereShape".
+
+First, wrap these UI elements in their own sub-widget, set the layout, and add it to the main widget. 
+
+1. To start, instantiate a `QWidget` called `entityNameWidget` and a `QFormLayout` called `formLayout`.
+   
+2. Later, you will create the input field and check box and add them to this widget. 
+
+3. Finally, set the layout of `entityNameWidget` to `formLayout`, and add `entityNameWidget` to `mainLayout`. 
+
+```cpp
+QWidget* entityNameWidget = new QWidget(this);
+QFormLayout* formLayout = new QFormLayout();
+
+// Create an input field and check box
+
+entityNameWidget->setLayout(formLayout);
+mainLayout->addWidget(entityNameWidget);
+```
+
+### Add an input field
+
```

Comment:
I would love to see screenshots under this headings of what the heading is referring to. E.g. Input field.